plugins/SitesManager/SiteContentDetection/Cloudflare.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link    https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Plugins\SitesManager\SiteContentDetection;
+
+use Piwik\Piwik;
+use Piwik\Request;
+use Piwik\SiteContentDetector;
+use Piwik\View;
+
+class Cloudflare extends SiteContentDetectionAbstract
+{
+    public static function getName(): string
+    {
+        return 'Cloudflare';
```

Comment:
It might be nice to have the Name, the Type, the Instruction Url and the Priority in protected static variables and then have the methods in the abstract class returning the values, so the actual detection classes don't have to have all the methods, just the config.

See for an example https://3v4l.org/JKtEk#v7.2.0

Reply:
I see now that that approach wouldn't support translations for technologies with names that can be/are usually translated like GTM.

Reply:
The problem with protected static variables is, that they can't be defined as abstract. So you can't force a sub class to overwrite the property. Which is why I used abstract methods instead. This makes it at lot easier to create subclasses without forgetting to define something that is required

Reply:
Hmmm, also an interface wouldn't help here. All good.